app/controllers/user_api_keys_controller.rb

```diff
@@ -76,17 +77,24 @@ def create
 
     # we keep the payload short so it encrypts easily with public key
     # it is often restricted to 128 chars
-    payload = {
+    @payload = {
       key: key.key,
       nonce: params[:nonce],
       push: key.has_push?,
       api: AUTH_API_VERSION
     }.to_json
 
     public_key = OpenSSL::PKey::RSA.new(params[:public_key])
-    payload = Base64.encode64(public_key.public_encrypt(payload))
-
-    redirect_to "#{params[:auth_redirect]}?payload=#{CGI.escape(payload)}"
+    @payload = Base64.encode64(public_key.public_encrypt(@payload))
+
+    (redirect_to("#{params[:auth_redirect]}?payload=#{CGI.escape(@payload)}") && return) if params[:auth_redirect]
```

Comment:
style wise ... the && return is just so easy to miss ... 

if params[:auth_redirect]
   ... do one thing
else
   ... do another thing
end